### PR TITLE
GEN-46 Driver script to launch subprocesses & basic server implementation

### PR DESCRIPTION
**Pull request by Ali-Aboelela**
Main.py opens the browser to http://localhost:5000, in the future this should be modified to where the React App is hosted. After this it calls two separate functions. These are blocking functions so 2 separate processes are instantiated to handle this.

The first function runs the server, the second function runs the react app.

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    # Example: Return the received data as JSON
+    response = {'message': 'Received string:', 'data': data}
+    return jsonify(response), 200
+
+
+def start_server():
+    serve(app, host='127.0.0.1', port=5000)
+
+if __name__ == '__main__':
```
Comment (NelsonLee-Code) [inline server.py:25]: This doesn't do anything, is this intended for the Main?

File: main.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    webbrowser.open_new_tab('http://localhost:5000')
+    
+    # Bind the first process to the run_server function
+    server_process = multiprocessing.Process(target=run_server)
+    
+    # Bind the second process to the run_app function
+    app_process = multiprocessing.Process(target=run_app)
+
+    # Start the server & app process
+    server_process.start()
```
Comment (NelsonLee-Code) [inline main.py:26]: Perhaps we can add a Try/Except w logging:

```import logging

logging.basicConfig(level=logging.DEBUG)

try:
    webbrowser.open_new_tab('http://localhost:5000')
except Exception as e:
    logging.error(f"Error opening browser: {e}")```

File: server.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from waitress import serve
+
+app = Flask(__name__)
+
+@app.route('/')
+def hello():
+    return "Hello, world!"
+
+@app.route('/query', methods=['POST'])
+def query_endpoint():
```
Comment (NelsonLee-Code) [inline server.py:11]: Similarly, add some Try/Except to ensure logging.

**Review by NelsonLee-Code — APPROVED** [review]
Implementing some Try/Except would be good for robustness of code.

File: server.py (hunk 1 above)
Comment (AliAboelela123) [reply server.py:25]: Yea this can just be deleted

**Pull request merged**